alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/utils/LazyMutable.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.model.implementations.utils
+
+import java.util.Optional
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+/**
+ * A delegate allowing to lazily initialise a mutable variable (= var).
+ */
+class LazyMutable<T>(private val initializer: () -> T) : ReadWriteProperty<Any?, T> {
+
+    private var value: Optional<T> = Optional.empty()
```

Comment:
Prefer `T?`, that's entirely internal.
Plus, this one is a candidate for being exported into a separate library. Can you open an issue about, once the PR gets merged?

Reply:
Yes